spark/src/main/scala/geotrellis/spark/reproject/TileRDDReproject.scala

```diff
@@ -187,37 +187,29 @@ object TileRDDReproject {
       val crs = rdd.metadata.crs
       val layout = rdd.metadata.layout
       val tileLayout = rdd.metadata.layout.tileLayout
-
-      val rasterExtents: RDD[(K, (RasterExtent, RasterExtent))] =
-        rdd
-          .mapPartitions({ partition =>
-            val transform = Transform(crs, destCrs)
-
-            partition.map { case (key, _) =>
-              val extent = key.getComponent[SpatialKey].extent(layout)
-              val rasterExtent = RasterExtent(extent, tileLayout.tileCols, tileLayout.tileRows)
-              (key, (rasterExtent, ReprojectRasterExtent(rasterExtent, transform)))
-            }
-          }, preservesPartitioning = true)
-
-      val borderSizesPerKey =
-        rasterExtents
-          .mapValues { case (re1, re2) =>
-            // Reproject the extent back into the original CRS,
-            // to determine how many border pixels we need.
-            // Pad by one extra pixel.
-            val e = re2.extent.reproject(destCrs, crs)
-            val gb = re1.gridBoundsFor(e, clamp = false)
-            BufferSizes(
-              left = 1 + (if(gb.colMin < 0) -gb.colMin else 0),
-              right = 1 + (if(gb.colMax >= re1.cols) gb.colMax - (re1.cols - 1) else 0),
-              top = 1 + (if(gb.rowMin < 0) -gb.rowMin else 0),
-              bottom = 1 + (if(gb.rowMax >= re1.rows) gb.rowMax - (re1.rows - 1) else 0)
-            )
-          }
-
-      val bufferedTiles =
-        rdd.bufferTiles(borderSizesPerKey)
+      // Avoid capturing instantiated Transform in closure because its not Serializable
+      lazy val transform = Transform(crs, destCrs)
```

Comment:
This is a trick worth calling out for smells (I think its inline with normal level of trickery).

Having a `lazy val` means that we're shipping only a `null` reference and a thunk as the value of the `transform`. Then each executor JVM will init its own instance of `transform`, once, and re-use it in the RDD evaluation. Thus this devilry avoids dealing with `TaskNotSerializable` exception and also allows re-use of heavy weight object per partition.